pkg/alertmanager/alertmanager.go

```diff
@@ -0,0 +1,150 @@
+package alertmanager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/alertmanager/config"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+
+	pkgconfig "github.com/giantswarm/observability-operator/pkg/config"
+)
+
+const (
+	// Those values are used to retrieve the Alertmanager configuration from the secret named after conf.Monitoring.AlertmanagerSecretName
+	// alertmanagerConfigKey is the key to the alertmanager configuration in the secret
+	alertmanagerConfigKey = "alertmanager.yaml"
+	// templatesSuffix is the suffix used to identify the templates in the secret
+	templatesSuffix = ".tmpl"
+
+	orgIDHeader         = "X-Scope-OrgID"
```

Comment:
I think we have this already for grafana orgs, maybe we should make it a common const instead?

Reply:
Yep sure, can do that